framework/src/meshgenerators/BlockToMeshConverterGenerator.C

```diff
@@ -0,0 +1,125 @@
+
+  auto new_mesh = buildMeshBaseObject();
+
+  std::vector<subdomain_id_type> target_block_ids =
+      MooseMeshUtils::getSubdomainIDs((*mesh), _target_blocks);
+
+  // Check that the block ids/names exist in the mesh
+  std::set<SubdomainID> mesh_blocks;
+  mesh->subdomain_ids(mesh_blocks);
+
+  for (std::size_t i = 0; i < target_block_ids.size(); ++i)
+    if (target_block_ids[i] == Moose::INVALID_BLOCK_ID || !mesh_blocks.count(target_block_ids[i]))
+    {
+      paramError("target_blocks",
+                 "The target_block '",
+                 getParam<std::vector<SubdomainName>>("target_blocks")[i],
+                 "' was not found within the mesh.");
+    }
+
+  // know which nodes have already been inserted, by tracking the old mesh's node's ids'
+  std::unordered_map<dof_id_type, dof_id_type> old_new_node_map;
+
+  for (subdomain_id_type target_block_id : target_block_ids)
+  {
+
+    for (auto elem : mesh->active_subdomain_elements_ptr_range(target_block_id))
+    {
+      // make a deep copy so that mutiple meshs' destructors don't segfault at program termination
+      auto copy = elem->build(elem->type());
+
+      // index of node in the copy element must be managed manually as there is no intelligent
+      // insert method
+      dof_id_type copy_n_index = 0;
+
+      // correctly assign new copies of nodes, loop over nodes
+      for (dof_id_type i : elem->node_index_range())
+      {
+        auto & n = *elem->node_ptr(i);
+
+        if (old_new_node_map.count(n.id()))
+        {
+          // case where we have already inserted this particular point before
+          // then we need to find the already-inserted one and hook it up right
+          // to it's respective element
+          copy->set_node(copy_n_index++) = new_mesh->node_ptr(old_new_node_map[n.id()]);
+        }
+        else
+        {
+          // case where we've NEVER inserted this particular point before
+          // add them both to the element and the mesh
+
+          // Nodes' IDs are their indexes in the nodes' respective mesh
+          // If we set them as invalid they are automatically re-assigned
+          // to be the last element in the mesh's node array
+          std::unique_ptr<Node> node_copy = Node::build(elem->point(i), Node::invalid_id);
+
+          // Add to mesh. This method call will automatically re-assign the id; so must call this
+          // first
+          new_mesh->add_node(node_copy.get());
+
+          // Add to element copy (manually)
+          copy->set_node(copy_n_index++) = node_copy.get();
+
+          // remember the (old) ID
+          old_new_node_map[n.id()] = node_copy.release()->id();
+        }
+      }
+
+      // it is ok to release the copy element into the mesh because derived meshes class
+      // (ReplicatedMesh, DistributedMesh) manage their own elements, will delete them
+      new_mesh->add_elem(copy.release());
```

Comment:
Just `add_elem(copy)`; libMesh will take a `unique_ptr` here.

In general, using `unique_ptr::release` is a code smell; it discards the safety guarantees you would otherwise get by using smart pointers as intended.

Reply:
I checked out a copy of the branch and made the change to see if it would work

```

In file included from /Users/savatc/projects/moose_v2/framework/build/unity_src/meshgenerators_Unity.C:5:
/Users/savatc/projects/moose_v2/framework/src/meshgenerators/BlockToMeshConverterGenerator.C:119:26: error: call to implicitly-deleted copy constructor of 'std::unique_ptr<Elem>'
      new_mesh->add_elem(copy);
                         ^~~~

```

It appears that the copy constructor of the unique_ptr class has been purposefully deleted, to prevent people from making a copy of a unique_ptr. This is to reinforce the notion that there should only be one unique_ptr pointing at an object, I reckon. 

The reason I `release()`-ed the unique_ptr here was because the unique would pass out of scope on the next line (120), and the the destructor of the unique_ptr would be called, which would delete the `Elem` object I had just added to the mesh. Then the mesh would have a bunch of null pointers to `Elem`s that no longer existed. 

The `release` method returns, not another unique_ptr --  but a raw pointer of what the unique_ptr is owning.

Reply:
Sorry, I meant to say `add_elem(std::move(copy))`.  You're right, you do have to explicitly tell the compiler that you're doing a move rather than a copy of the parameter.